File: slinding window/max_no_of_vowels_in_substring_of_given_length.java
public class max_no_of_vowels_in_substring_of_given_length {
    
}
// tc O(n), sc O(1)
class Solution {
    public int maxVowels(String s, int k) {
        int l=0, res=0, numOfVowels=0;
        for(int r=0;r<s.length();r++){
            if("aeiou".indexOf(s.charAt(r))!=-1){
                numOfVowels++;
            }
            if((r-l+1)==k){
                res = Math.max(res, numOfVowels);
                if("aeiou".indexOf(s.charAt(l))!=-1){
                    numOfVowels--;
                }
                l++;
            }
        }
        return res;
    }
}

File: slinding window/max_no_of_vowels_in_substring_of_given_length.py
# sliding window
# tc O(n), sc O(1)
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        l = 0
        numOfVowels = 0
        res = 0
        for r in range(len(s)):
            if s[r] in "aeiou":
                numOfVowels += 1
            if (r - l + 1) == k:
                res = max(res, numOfVowels)
                if s[l] in "aeiou":
                    numOfVowels -= 1
                l += 1
        return res
